wallet/tests/e2e/specs/e2e-spec.js

```diff
@@ -0,0 +1,252 @@
+      cy.get('button').contains('Transfer').click();
+      cy.get('button').contains('Create Transaction').click();
+      cy.get('#Bridge_modal_continueTransferButton').click();
+      cy.wait(30000);
+      cy.confirmMetamaskTransaction().then(confirmed => expect(confirmed).to.be.true);
+      cy.wait(50000);
+      cy.get('.btn-close').click();
+      cy.contains('Nightfall Assets').click();
+      cy.wait(20000);
+    });
+
+    it(`check token balance equal to ${depositValue * 4}`, () => {
+      cy.get('#TokenItem_tokenBalanceMATIC').should($div => {
+        const totalBalance = Number($div.text());
+        expect(totalBalance).to.equal(depositValue * 4);
+        currentTokenBalance = totalBalance;
+      });
+    });
+  });
+
+  context('Withdraw', () => {
+    const withdrawValue = 4;
+
+    it(`withdraw token of value ${withdrawValue}`, () => {
+      cy.get('#TokenItem_tokenWithdrawMATIC').click();
+      // cy.get('#Bridge_amountDetails_tokenAmount').clear().type(withdrawValue);
+      cy.get('label').contains('Withdraw').click();
+      cy.get('#Bridge_amountDetails_tokenAmount').type(withdrawValue);
+      cy.get('button').contains('Transfer').click();
+      cy.get('button').contains('Create Transaction').click();
+      cy.get('#Bridge_modal_continueTransferButton').click();
+      cy.wait(30000);
+      cy.confirmMetamaskTransaction().then(confirmed => expect(confirmed).to.be.true);
+      cy.wait(50000);
+      cy.get('.btn-close').click();
+      cy.contains('Nightfall Assets').click();
+    });
+
+    it(`check token balance after withdraw`, () => {
+      cy.get('#TokenItem_tokenBalanceMATIC').should($div => {
+        const totalBalance = Number($div.text());
+        expect(totalBalance).to.equal(currentTokenBalance - withdrawValue);
+        currentTokenBalance = totalBalance;
+      });
+    });
+  });
+
+  context('Single Transfer', () => {
+    const transferValue = 4;
+
+    /*
+     * dummy pkd of user who does not exist
+     * Note: even though we are passing recipientPkd but in code
+     *   it gets override by sender's pdk for now
+     *   hence two check
+     *   check1: before Block Proposed event
+     *   check2: after Block Proposed event
+     */
+    const recipientPkd = '0x90ff185f7fa35ddae731ddad18a958d55d45bb973c16735018f6bc6f3798a7e1';
+
+    it(`transfer token of value ${transferValue}`, () => {
+      cy.get('#TokenItem_tokenSendMATIC').click();
+      cy.get('#TokenItem_modalSend_tokenAmount').clear().type(transferValue);
+      cy.get('#TokenItem_modalSend_compressedPkd').clear().type(recipientPkd);
+      cy.get('button').contains('Continue').click();
+      cy.wait(50000);
+      cy.confirmMetamaskTransaction().then(confirmed => expect(confirmed).to.be.true);
+      cy.get('.btn-close').click();
+      cy.contains('L2 Bridge').click();
+      cy.contains('Nightfall Assets').click();
+    });
+
+    // check1
+    it(`check token balance after transfer - before block proposed event`, () => {
```

Comment:
The test should be written from the perspective of user who will not track block proposed events or know the size of L2 blocks. They can only check if their balance is pending or finalised. 
References on how to test without considering block proposed in `test/ping-pong/user-local/src/index.mjs` and `test/adversary.test.mjs`